apps/directory/src/filter-config/filterOptions.js

```diff
@@ -136,36 +136,40 @@ export const ontologyFilterOptions = (filterFacet) => {
       ];
 
       if (!cachedOptions.length) {
-        new queryEMX2("graphql")
-          .table(sourceTable)
-          .select(selection)
-          .orderBy(sourceTable, sortColumn, sortDirection)
-          .execute()
-          .then((response) => {
-            const onlyParents = response[sourceTable].filter(
-              (row) => !row.parent
-            );
-            const itemsSplitByOntology = {};
-
-            for (const ontologyItem of onlyParents) {
-              for (const ontologyId of ontologyIdentifiers) {
-                if (
-                  ontologyItem.name
-                    .toLowerCase()
-                    .includes(ontologyId.toLowerCase())
-                ) {
-                  if (!itemsSplitByOntology[ontologyId]) {
-                    itemsSplitByOntology[ontologyId] = [ontologyItem];
-                  } else {
-                    itemsSplitByOntology[ontologyId].push(ontologyItem);
+        /** make it query after all the others, saves 50% of initial load */
+        const waitAfterBiobanks = setTimeout(() => {
```

Comment:
no way to do this with a promise instead of a timeout?

Reply:
no because of the async loading of the store. a promise will stop the execution of the code BUT will query. the timeout stops the execution of the code _before_ the query